saas_demo/os.py

```diff
@@ -120,6 +125,84 @@ def update_config(section, key, value):
         config_parser.write(configfile)
 
 
+def _fileopen(path, mode, basedir, pathinfo):
+    name = os.path.normpath(os.path.normcase(os.path.join(basedir, path)))
+    if os.path.isfile(name):
+        if 'b' in mode:
+            fo = open(name, mode)
+        else:
+            fo = io.open(name, mode, encoding='utf-8')
+        if pathinfo:
+            return fo, name
+        return fo
+
+
+def file_open(name, mode="r", pathinfo=False):
+    if os.path.isabs(name):
+        base, name = os.path.split(name)
+        return _fileopen(name, mode=mode, basedir=base, pathinfo=pathinfo)
+
+
+def adapt_version(version):
+    serie = release.major_version
+    if version == serie or not version.startswith(serie + '.'):
```

Comment:
What about version ``11.0.1`` ?

Reply:
This is also a copied method, but copied by mistake, we can get rid of it and just import

Reply:
Consider to make a PR to Odoo with fixing that